src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllManStatement
+                         = connection.prepareStatement(getAllManRequest)) {
+            ResultSet resultSet = getAllManStatement.executeQuery();
+            List<Manufacturer> manufacturersList = new ArrayList<>();
+            while (resultSet.next()) {
+                manufacturersList.add(getManFromResultSet(resultSet));
+            }
+            return manufacturersList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManRequest = "UPDATE manufacturers SET name = ?, "
+                + "country = ? WHERE is_deleted = false AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateManStatement
+                         = connection.prepareStatement(updateManRequest)) {
+            updateManStatement.setString(1, manufacturer.getName());
+            updateManStatement.setString(2, manufacturer.getCountry());
+            updateManStatement.setLong(3, manufacturer.getId());
+            if (updateManStatement.executeUpdate() > 0) {
+                return manufacturer;
+            } else {
+                throw new DataProcessingException("Can't update manufacturer in DB: "
+                        + manufacturer, new SQLException("id was not found or deleted"));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer in DB: "
+                    + manufacturer, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteManRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteManStatement
+                         = connection.prepareStatement(deleteManRequest)) {
+            deleteManStatement.setLong(1, id);
+            return deleteManStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer from DB by id: " + id, e);
+        }
+    }
+
+    private Manufacturer getManFromResultSet(ResultSet resultSet) {
+        Manufacturer manufacturer = new Manufacturer();
+        try {
+            manufacturer.setId(resultSet.getObject("id", Long.class));
+            manufacturer.setName(resultSet.getString("name"));
+            manufacturer.setCountry(resultSet.getString("country"));
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get information from ResultSet: "
+                    + resultSet, new SQLException());
```

Comment:
```suggestion
            throw new DataProcessingException("Can't get information from ResultSet: "
                    + resultSet, e);
```

we need to pass info about existing exception